air-quality-ui/system_tests/ui/city_page/city_page_api_calls.spec.ts

```diff
@@ -107,8 +112,86 @@ test.describe('API calls on changing forecast base time in UI', () => {
       )
 
       await banner.clickOnDay(3)
-      await banner.confirmDate()
+      await banner.windowUpdateClick()
       expect(requestArray[0]).toContain(`base_time=${expectedForecastBaseTime}`)
     })
   })
 })
+
+test.describe('Forecast window for city page', () => {
+  let forecastRequestArray: string[]
+  let measurementsRequestArray: string[]
+  test.beforeEach(async ({ page, cityPage, basePage, banner, summaryPage }) => {
+    await gotoPage(page, '/city/Rio%20de%20Janeiro')
+    await cityPage.waitForAllGraphsToBeVisible()
+    await cityPage.setBaseTime('01/07/2024 00:00')
+
+    forecastRequestArray = await summaryPage.captureNetworkRequestsAsArray(
+      page,
+      httpMethodGet,
+      basePage.baseAPIURL + forecastAPIEndpoint,
+    )
+    measurementsRequestArray = await summaryPage.captureNetworkRequestsAsArray(
+      page,
+      httpMethodGet,
+      basePage.baseAPIURL + measurementsAPIEndpoint,
+    )
+    await banner.forecastWindowDropdownClick()
+  })
+  const testCases = [
+    { windowOption: '1', days: 1, toDate: '2024-07-02T00:00:00Z' },
+    { windowOption: '2', days: 2, toDate: '2024-07-03T00:00:00Z' },
+    { windowOption: '3', days: 3, toDate: '2024-07-04T00:00:00Z' },
+    { windowOption: '4', days: 4, toDate: '2024-07-05T00:00:00Z' },
+    { windowOption: '5', days: 5, toDate: '2024-07-06T00:00:00Z' },
+  ]
+  test.describe('Forecast API array', () => {
+    for (const { windowOption, days, toDate } of testCases) {
+      test(`Forecast window ${windowOption} requests ${days} day(s) worth of data`, async ({
+        banner,
+        page,
+        cityPage,
+      }) => {
+        const expectedValidTimeFrom = await encodeDateToURIComponent(
+          new Date('2024-07-01T00:00:00Z'),
+        )
+        const expectedValidTimeTo = await encodeDateToURIComponent(
+          new Date(toDate),
+        )
+
+        await banner.forecastWindowDropdownSelect(windowOption)
+        await banner.windowUpdateClick()
+        await waitForIdleNetwork(page, cityPage.aqiChart)
+
+        await expect(forecastRequestArray[0]).toContain(
+          `valid_time_from=${expectedValidTimeFrom}&valid_time_to=${expectedValidTimeTo}`,
```

Comment:
shall we assert the rest of the request in here too eg forecast base time, location name, location type as this is not recorded elsewhere anyway? Similarly for the other tests that assert on the call url